src/cm_adapter.ts

```diff
@@ -547,7 +564,21 @@ export class CodeMirror {
       }
     }
 
-    return posFromIndex(doc, range.head);
+    let pos = posFromIndex(doc, range.head) as Pos&{hitSide:boolean};
+    if (
+      (
+        amount < 0 && 
+        range.head == 0 && goalColumn != 0 &&
+        start.line == 0 && start.ch != 0
+      ) || (
+        amount > 0 && 
+        range.head == doc.length && pos.ch != goalColumn
+        && start.line == pos.line
+      )
+    ) {
+      pos.hitSide = true;
+    }
+    return pos;
```

Comment:
Added a comment. This is needed to prevent gj/gk from moving to document edge https://github.com/replit/codemirror-vim/blob/e163b3f29ef50a944408eab03f85632b68db7778/src/vim.js#L1964